diff -ur laser_geometry-release-release-noetic-laser_geometry-1.6.5-1.orig/CMakeLists.txt laser_geometry-release-release-noetic-laser_geometry-1.6.5-1/CMakeLists.txt
--- laser_geometry-release-release-noetic-laser_geometry-1.6.5-1.orig/CMakeLists.txt	2020-03-13 21:10:53.000000000 +0100
+++ laser_geometry-release-release-noetic-laser_geometry-1.6.5-1/CMakeLists.txt	2020-05-24 21:10:08.541354599 +0200
@@ -17,7 +17,7 @@
 catkin_python_setup()
 
 catkin_package(
-  INCLUDE_DIRS include
+  INCLUDE_DIRS include ${EIGEN3_INCLUDE_DIR}
   LIBRARIES laser_geometry
   CATKIN_DEPENDS tf tf2
   DEPENDS Boost EIGEN3
