commit ded9280f9ef5157d5e057ce386a7b3a79a17b56b
Author: Till Hofmann <hofmann@kbsg.rwth-aachen.de>
Date:   Mon May 14 17:11:53 2018 +0200

    Replace all unversioned python shebangs by python2 shebangs

diff --git a/bin/catkin_find b/bin/catkin_find
index c3c7b22..784390e 100755
--- a/bin/catkin_find
+++ b/bin/catkin_find
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_init_workspace b/bin/catkin_init_workspace
index e7b7552..dd62fdd 100755
--- a/bin/catkin_init_workspace
+++ b/bin/catkin_init_workspace
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_make b/bin/catkin_make
index d02ced0..78ac4f4 100755
--- a/bin/catkin_make
+++ b/bin/catkin_make
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_make_isolated b/bin/catkin_make_isolated
index 1642075..60e7445 100755
--- a/bin/catkin_make_isolated
+++ b/bin/catkin_make_isolated
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_package_version b/bin/catkin_package_version
index 422b5f6..fb007ba 100755
--- a/bin/catkin_package_version
+++ b/bin/catkin_package_version
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_prepare_release b/bin/catkin_prepare_release
index 2a38bba..a7748ea 100755
--- a/bin/catkin_prepare_release
+++ b/bin/catkin_prepare_release
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_test_results b/bin/catkin_test_results
index 8a1686e..648e79d 100755
--- a/bin/catkin_test_results
+++ b/bin/catkin_test_results
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/bin/catkin_topological_order b/bin/catkin_topological_order
index a6776d3..158aaf3 100755
--- a/bin/catkin_topological_order
+++ b/bin/catkin_topological_order
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import argparse
diff --git a/cmake/interrogate_setup_dot_py.py b/cmake/interrogate_setup_dot_py.py
index 02954de..56bd440 100644
--- a/cmake/interrogate_setup_dot_py.py
+++ b/cmake/interrogate_setup_dot_py.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Software License Agreement (BSD License)
 #
diff --git a/cmake/order_paths.py b/cmake/order_paths.py
index d2a3eca..6caf913 100644
--- a/cmake/order_paths.py
+++ b/cmake/order_paths.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 
diff --git a/cmake/parse_package_xml.py b/cmake/parse_package_xml.py
index d6c9e21..c18fe76 100755
--- a/cmake/parse_package_xml.py
+++ b/cmake/parse_package_xml.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Software License Agreement (BSD License)
 #
diff --git a/cmake/templates/_setup_util.py.in b/cmake/templates/_setup_util.py.in
index f696ba1..07d338a 100755
--- a/cmake/templates/_setup_util.py.in
+++ b/cmake/templates/_setup_util.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 # Software License Agreement (BSD License)
diff --git a/cmake/templates/script.py.in b/cmake/templates/script.py.in
index 0d64646..5ef937a 100755
--- a/cmake/templates/script.py.in
+++ b/cmake/templates/script.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # generated from catkin/cmake/template/script.py.in
 # creates a relay to a python script source file, acting as that file.
diff --git a/cmake/test/download_checkmd5.py b/cmake/test/download_checkmd5.py
index 7394315..e663a0b 100755
--- a/cmake/test/download_checkmd5.py
+++ b/cmake/test/download_checkmd5.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 import errno
diff --git a/cmake/test/remove_test_results.py b/cmake/test/remove_test_results.py
index 5a864a5..d466c89 100755
--- a/cmake/test/remove_test_results.py
+++ b/cmake/test/remove_test_results.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 
diff --git a/cmake/test/run_tests.py b/cmake/test/run_tests.py
index 9b93d55..c8a36c9 100755
--- a/cmake/test/run_tests.py
+++ b/cmake/test/run_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from __future__ import print_function
 
diff --git a/doc/generate_cmake_rst.py b/doc/generate_cmake_rst.py
index 8edeaa3..701059d 100755
--- a/doc/generate_cmake_rst.py
+++ b/doc/generate_cmake_rst.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Software License Agreement (BSD License)
 #
diff --git a/doc/howto/format1/installing_python.rst b/doc/howto/format1/installing_python.rst
index 148c4f4..8a4ef3e 100644
--- a/doc/howto/format1/installing_python.rst
+++ b/doc/howto/format1/installing_python.rst
@@ -35,7 +35,7 @@ Another good practice is to keep executable scripts very short,
 placing most of the code in a module which the script imports and then
 invokes::
 
-  #! /usr/bin/env python
+  #! /usr/bin/env python2
   import your_package.main
   if __name__ == '__main__':
       your_package.main()
diff --git a/doc/howto/format2/installing_python.rst b/doc/howto/format2/installing_python.rst
index 2fcee44..01ff7ff 100644
--- a/doc/howto/format2/installing_python.rst
+++ b/doc/howto/format2/installing_python.rst
@@ -35,7 +35,7 @@ Another good practice is to keep executable scripts very short,
 placing most of the code in a module which the script imports and then
 invokes::
 
-  #! /usr/bin/env python
+  #! /usr/bin/env python2
   import your_package.main
   if __name__ == '__main__':
       your_package.main()
diff --git a/setup.py b/setup.py
index 4674e3c..c4973fd 100644
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from distutils.core import setup
 from catkin_pkg.python_setup import generate_distutils_setup
diff --git a/test/checks/fuerte/distro-to-install.py b/test/checks/fuerte/distro-to-install.py
index ecc5775..8aec10b 100755
--- a/test/checks/fuerte/distro-to-install.py
+++ b/test/checks/fuerte/distro-to-install.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import rospkg.distro
 import sys
diff --git a/test/local_tests/test_with_mock_workspace.py b/test/local_tests/test_with_mock_workspace.py
index 29fe8c9..7c6735d 100644
--- a/test/local_tests/test_with_mock_workspace.py
+++ b/test/local_tests/test_with_mock_workspace.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os
 import shutil
diff --git a/test/network_tests/test_basics.py b/test/network_tests/test_basics.py
index c71bd7d..5e94be1 100644
--- a/test/network_tests/test_basics.py
+++ b/test/network_tests/test_basics.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os
 import sys
diff --git a/test/network_tests/test_unstable.py b/test/network_tests/test_unstable.py
index e6437b8..47ee409 100644
--- a/test/network_tests/test_unstable.py
+++ b/test/network_tests/test_unstable.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os
 import shutil
