diff -ur geometry-release-release-noetic-tf-1.13.2-1/CMakeLists.txt geometry-release-release-noetic-tf-1.13.2-1_new/CMakeLists.txt
--- geometry-release-release-noetic-tf-1.13.2-1/CMakeLists.txt	2020-06-09 08:29:32.000000000 +0200
+++ geometry-release-release-noetic-tf-1.13.2-1_new/CMakeLists.txt	2022-12-25 10:10:57.889426857 +0100
@@ -3,15 +3,8 @@
 
 include(CheckCXXCompilerFlag)
 unset(COMPILER_SUPPORTS_CXX11 CACHE)
-if(MSVC)
-  # https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version
-  # MSVC has c++14 enabled by default, no need to specify c++11
-else()
-  check_cxx_compiler_flag(-std=c++11 COMPILER_SUPPORTS_CXX11)
-  if(COMPILER_SUPPORTS_CXX11)
-    add_compile_options(-std=c++11)
-  endif()
-endif()
+
+add_compile_options(-std=c++17)
 
 find_package(catkin REQUIRED COMPONENTS
     angles
