diff -ur perception_pcl-release-release-noetic-pcl_ros-1.7.4-1/CMakeLists.txt perception_pcl-release-release-noetic-pcl_ros-1.7.4-1_new/CMakeLists.txt
--- perception_pcl-release-release-noetic-pcl_ros-1.7.4-1/CMakeLists.txt	2022-02-09 02:01:56.000000000 +0100
+++ perception_pcl-release-release-noetic-pcl_ros-1.7.4-1_new/CMakeLists.txt	2022-12-25 18:30:32.473850404 +0100
@@ -5,9 +5,9 @@
 # Use CMake C++ standard management where possible for better interoperability
 # with dependencies that export target standards and/or features
 if (${CMAKE_VERSION} VERSION_LESS "3.1")
-  add_compile_options(-std=c++14)
+  add_compile_options(-std=c++17)
 else()
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 17)
 endif()
 
 ## Find catkin packages
