diff -ur ros_comm-release-release-melodic-roscpp-1.14.3-0/CMakeLists.txt ros_comm-release-release-melodic-roscpp-1.14.3-0.signals2/CMakeLists.txt
--- ros_comm-release-release-melodic-roscpp-1.14.3-0/CMakeLists.txt	2018-08-06 19:38:50.000000000 +0200
+++ ros_comm-release-release-melodic-roscpp-1.14.3-0.signals2/CMakeLists.txt	2019-07-14 09:09:18.983412338 +0200
@@ -22,7 +22,7 @@
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/include/ros/common.h.in ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}/ros/common.h)
 
-find_package(Boost REQUIRED COMPONENTS chrono filesystem signals system)
+find_package(Boost REQUIRED COMPONENTS chrono filesystem system)
 
 include_directories(include ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_INCLUDE_DESTINATION}/ros ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 
