diff -ur image_common-release-release-melodic-camera_calibration_parsers-1.11.13-0/CMakeLists.txt image_common-release-release-melodic-camera_calibration_parsers-1.11.13-0.python/CMakeLists.txt
--- image_common-release-release-melodic-camera_calibration_parsers-1.11.13-0/CMakeLists.txt	2018-03-26 23:42:19.000000000 +0200
+++ image_common-release-release-melodic-camera_calibration_parsers-1.11.13-0.python/CMakeLists.txt	2019-07-23 13:41:03.711978770 +0200
@@ -4,11 +4,7 @@
 find_package(catkin REQUIRED sensor_msgs rosconsole roscpp roscpp_serialization)
 
 find_package(PythonLibs REQUIRED)
-if(PYTHONLIBS_VERSION_STRING VERSION_LESS 3)
-  find_package(Boost REQUIRED COMPONENTS filesystem python)
-else()
-  find_package(Boost REQUIRED COMPONENTS filesystem python3)
-endif()
+find_package(Boost REQUIRED COMPONENTS filesystem python${PYTHON_VERSION_NODOTS})
 include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIRS})
 
 catkin_python_setup()
