diff -ur vision_opencv-release-release-melodic-cv_bridge-1.13.0-0/CMakeLists.txt vision_opencv-release-release-melodic-cv_bridge-1.13.0-0.python/CMakeLists.txt
--- vision_opencv-release-release-melodic-cv_bridge-1.13.0-0/CMakeLists.txt	2018-04-30 19:54:22.000000000 +0200
+++ vision_opencv-release-release-melodic-cv_bridge-1.13.0-0.python/CMakeLists.txt	2019-07-23 13:45:57.189255727 +0200
@@ -5,11 +5,7 @@
 
 if(NOT ANDROID)
   find_package(PythonLibs)
-  if(PYTHONLIBS_VERSION_STRING VERSION_LESS 3)
-    find_package(Boost REQUIRED python)
-  else()
-    find_package(Boost REQUIRED python3)
-  endif()
+  find_package(Boost REQUIRED python${PYTHON_VERSION_NODOTS})
 else()
 find_package(Boost REQUIRED)
 endif()
